services/venice-router/src/main/java/com/linkedin/venice/router/streaming/VeniceChunkedResponse.java

```diff
@@ -274,40 +274,41 @@ public CompletableFuture<Long> write(
       ByteBuf byteBuf,
       CompressionStrategy compression,
       StreamingCallback<Long> callback) {
-    if (this.requestType.equals(RequestType.MULTI_GET_STREAMING)) {
-      if (this.responseCompression == null) {
-        synchronized (this) {
-          if (this.responseCompression == null) {
-            this.responseCompression = compression;
-            this.responseMetadata = RESPONSE_META_MAP_FOR_MULTI_GET.get(compression);
-          }
+    boolean isFirstWrite = false;
+    if (this.responseCompression == null) {
+      synchronized (this) {
+        if (this.responseCompression == null) {
+          this.responseCompression = compression;
+          isFirstWrite = true;
         }
       }
-      if (!this.responseCompression.equals(compression)) {
-        // Defensive code, not expected
-        LOGGER.error(
-            "Received inconsistent compression for the new write: {}, and previous compression: {}",
-            compression,
-            this.responseCompression);
-        /**
-         * Skip the write with wrong compression.
-         * {@link VeniceResponseAggregator#buildStreamingResponse} will perform the same check
-         * when all the responses returned by storage nodes are ready, and when the inconsistency happens,
-         * {@link VeniceResponseAggregator} will return an error response instead to indicate the issue.
-         */
-        return CompletableFuture.completedFuture(0l);
-      }
-    } else {
-      responseMetadata = RESPONSE_META_FOR_COMPUTE;
+    }
+
+    boolean isMultiGet = this.requestType.equals(RequestType.MULTI_GET_STREAMING);
```

Comment:
Can we rename it to be `isMultiGetStreaming`?

Reply:
Sure, done.